compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1056,9 +1056,21 @@ impl<'tcx> Predicate<'tcx> {
         }
     }
 
+    /// Converts this to a `Binder<PredicateAtom<'tcx>>`. If the value was an
+    /// `Atom`, then it is not allowed to contain escaping bound vars.
+    pub fn bound_atom(self, _tcx: TyCtxt<'tcx>) -> Binder<PredicateAtom<'tcx>> {
+        match self.kind() {
+            &PredicateKind::ForAll(binder) => binder,
+            &PredicateKind::Atom(atom) => {
+                assert!(!atom.has_escaping_bound_vars());
```

Comment:
```suggestion
                debug_assert!(!atom.has_escaping_bound_vars());
```
i expect this method to be faily hot, so it might also make sense to mark this `#[inline]`

Reply:
also, is there a reason you did not remove the `_tcx` argument here? Do you expect us to need it in the future?

Reply:
Oops, definitely meant to make this a `debug_assert`. I will hold off on adding `#[inline]` for now, since other functions don't have it. It's probably worth a followup.

As for `_tcx`, I can remove that. I don't see any need for it in the future, here.